s3/src/main/java/ch/cyberduck/core/sts/AssumeRoleWithWebIdentitySTSCredentialsConfigurator.java

```diff
@@ -43,12 +45,17 @@ public AssumeRoleWithWebIdentitySTSCredentialsConfigurator(final X509TrustManage
 
     @Override
     public Credentials configure(final Host host) {
-        final Credentials credentials = new Credentials(host.getCredentials());
+        final Credentials hostCredentials = new Credentials(host.getCredentials());
 
         service = this.getTokenService(host, null, null, null, null);
 
-        AssumeRoleWithWebIdentityRequest webIdReq = new AssumeRoleWithWebIdentityRequest()
-                .withWebIdentityToken(credentials.getOauth().getAccessToken());
+        AssumeRoleWithWebIdentityRequest webIdReq = new AssumeRoleWithWebIdentityRequest();
+        if(hostCredentials.getToken().isEmpty()) {
```

Comment:
The OAuth token should be obtained from the `OAuth2RequestInterceptor` which stores the current values (to make sure you get the updated token values after a refresh.